chain/chain/src/chain.rs

```diff
@@ -4247,6 +4234,48 @@ impl Chain {
         }
         Ok(())
     }
+
+    pub fn get_cached_state_parts(
```

Comment:
How do we keep an eye on how intensive is this operation? Do you think it would be significantly faster to store this cache in a new column and update it when the node caches new parts. Parts can be added either by generating them or downloading them.

Reply:
"Premature optimization is the root of all evil" (Knuth).

This PR fulfills the functional requirement. 
As I say in the PR description:
> Will follow up by benchmarking and maybe caching.

I will get to benchmarking after fulfilling the functional requirements to avoid overly complicated PRs.